### DISCUSSION TITLE
Search

### DISCUSSION THREAD

**Discussion by robertjmccarthy** [ministryofjustice/moj-frontend#712] [RELEASED: Components, pages and patterns]
The [search component](https://design-patterns.service.justice.gov.uk/components/search/) is live in the MoJ Design System.

Use this discussion to suggest changes and improvements to this component.

**Comment by robertjmccarthy** [comment]
@chrispymm commented on #618

_Standard GOV.UK design system text inputs allow a html option to be passed instead of text for the hint text. The search component does not allow this, and passing a hint with an html key simply renders an empty hint element.

For consistency with the main design system and ease of use for developers we should follow GOV.UK conventions and match where possible the options accepted by the GOV.UK text input._

#618 will be closed as this discussion will be reviewed when the search component is improved.

**Comment by helennickols** [comment]
GOV.UK launched an A/B test of an autocomplete feature on their site search. Details are in [this Slack post (internal to MoJ)](https://mojdt.slack.com/archives/C58FBN6V6/p1732276546382549). Findings from this test may be relevant for this MoJ DS component should this feature be considered for an iteration.

**Comment by helennickols** [comment]
There's a new, improved version of the search component on the MOJ Design System.

**Changes to the search component** 

The new version of the component:
- has 2 layouts (stacked and inline)
- has a magnifying glass icon button
- is WCAG 2.2 compliant  

Usage has been thoroughly documented.  

**Breaking changes** 

If you're using Nunjucks, you do not need to do anything.

If you're using HTML, you'll need to change your templates. View the [MOJ Frontend release notes](https://github.com/ministryofjustice/moj-frontend/releases/tag/v11.0.0). 

## Help and feedback 

You can [give feedback on the search component in Github](https://github.com/ministryofjustice/moj-frontend/discussions/712).
 
For help, contact the Design System team:

- on the [moj-design-system-support Slack channel](https://moj.enterprise.slack.com/archives/CH5RUSB27) (for MoJ staff)
- on the [moj-design-system Slack channel](https://ukgovernmentdigital.slack.com/archives/CJ6QDRDGC) (for all UK government staff) 
- by email [design-system@justice.gov.uk](mailto:design-system@justice.gov.uk) (for everyone)
